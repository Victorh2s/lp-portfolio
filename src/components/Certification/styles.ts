import styled, { css } from 'styled-components';

export const Certification = styled.div`
  ${({ theme }) => css`
    padding: ${theme.spacings.large};
    text-decoration: none;
    font-family: ${theme.font.family.main};
    margin-top: 0.6rem;


  @media ${theme.media.lteMedium}{
    > a{
      font-size: 1.4rem;
    }
  }

  `}
`;

export const a = styled.a`
  ${({ theme }) => css`

    >svg {
    width: 2rem;
    height: 2rem;
    color: ${theme.colors.green};
  }



  `}
`;
