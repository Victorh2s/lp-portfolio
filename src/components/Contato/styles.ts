import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
  align-items: center;
  padding: ${theme.spacings.xlarge};

  `}
`;

export const Nav = styled.nav`
display: flex;
  align-items: center;
  justify-content: center;
`;

export const Icons = styled.div`

`;

export const Container = styled.a`
  ${({ theme }) => css`

  color:${theme.colors.white};
  padding: ${theme.spacings.small};

  >svg{
    width: 5rem;
    height: 5rem;


  }

  @media ${theme.media.lteMedium}{
    >svg{
    width: 3rem;
    height: 3rem;
  }
  }

  `}
`;
