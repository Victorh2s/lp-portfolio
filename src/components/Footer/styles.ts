import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    background: ${theme.colors.dark};
    border-top: 0.1rem solid ${theme.colors.green};
    align-items: center;
    justify-content: center;
    padding: 1rem;
  `}
`;

export const Container = styled.a`
    border: 0.1rem solid;
    border-radius: 50rem;
    width: 4rem;
    height: 4rem;
    z-index: 6;
`;
