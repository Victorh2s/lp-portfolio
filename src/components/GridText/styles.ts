import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`

  display: grid;

  grid-template-columns: 2fr 2fr;
  align-items: stretch;
  justify-content: center;
  @media ${theme.media.lteMedium}{
      display: block;
      align-items: center;
      justify-content: center;
      margin: 0 auto;
  }

  `}
`;

export const Container = styled.div`
  ${({ theme }) => css`

  padding: ${theme.spacings.large};
  `}
`;
