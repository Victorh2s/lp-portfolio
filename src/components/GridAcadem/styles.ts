import { Wrapper as TextGrid } from '../TextGrid/styles';
import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';
import { Certification } from '../Certification/styles';

export const Grid = styled.main`
  ${({ theme }) => css`

  @media ${theme.media.lteMedium}{
    ${Heading}{
      font-size: ${theme.font.sizes.medium};
    }


}
  `}
`;

export const Container = styled.section`
  ${({ theme }) => css`

  ${TextGrid}{
    margin-top: -0.1rem;
  }

  @media ${theme.media.lteMedium}{
    ${TextGrid}{
    font-size: 1.4rem;
  }


}

  `}
`;

export const Section = styled.div`
  ${({ theme }) => css`
  display: flex;


  @media ${theme.media.lteMedium}{
      display: block;
      align-items:center;
      ${Certification}{
        margin-top: -0.5rem;
        margin-bottom: 1rem;
        padding: 0;
      }

  }

  `}
`;

export const Title = styled.div`
  ${({ theme }) => css`

  display: flex;
  align-items: center;
  ${Heading}{
    padding: 1rem;
    }



  @media ${theme.media.lteMedium}{
    ${Image}{
      width: 5rem;
      height: 4rem;
      padding: 0.1rem;
    }
    ${Heading}{
      font-size: 2rem;
      padding: 0;
    }

    display: inline;
  }


  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
  width: 6rem;
  height: 5rem;
  padding: 0.3rem;
  border-radius: 10%;
  background: ${theme.colors.white};

  `}
`;
