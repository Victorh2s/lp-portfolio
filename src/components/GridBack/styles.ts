import { Title as Heading } from '../Heading/styles';
import { Wrapper as TextGrid } from '../TextGrid/styles';
import styled, { css } from 'styled-components';

export const Front = styled.div`
  ${({ theme }) => css`
  padding: ${theme.spacings.xlarge};

${Heading}{
    text-align: center;
}

border: 0.1rem solid;
border-image-slice: 1;
border-width: 0.1rem;
border-image-source: linear-gradient(to right, #090213, #078a85);

@media ${theme.media.lteMedium}{
padding: ${theme.spacings.small};
}

  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(2, 2fr);
    gap: ${theme.spacings.small};



    @media ${theme.media.lteMedium}{
      display: block;
      ${TextGrid}{
        font-size: ${theme.font.sizes.small};
      }
      ${Image}{
        width: 7rem;
        height: 7rem;
      }
  }
  `}
`;

export const container = styled.div`
  ${({ theme }) => css`
text-align: center;
justify-content: center;

>svg{
  color:${theme.colors.white};
  width:10rem;
  height:10rem;
  border: 0.1rem solid;
  border-image-slice: 1;
  border-width: 0.1rem;
  border-image-source: linear-gradient(to left, #090213, #078a85);
  transition: all 200ms ease-in-out;
  padding: ${theme.spacings.small};
  cursor: pointer;

  &:hover{
  border: 0.1rem solid;
  border-image-slice: 1;
  border-width: 0.4rem;
  border-image-source: linear-gradient(to left, #090213, #078a85);
  }
}

${TextGrid}{
  margin-top: 0.8rem;
  border: 0.1rem solid;
  border-image-slice: 1;
  border-width: 0.1rem;
  border-image-source: linear-gradient(to right, #090213, #078a85);
}
`}
`;

export const Image = styled.img`
  width:10rem;
  height:10rem;
  border: 0.1rem solid;
  border-image-slice: 1;
  border-width: 0.1rem;
  border-image-source: linear-gradient(to left, #090213, #078a85);
  transition: all 200ms ease-in-out;

  cursor: pointer;
  &:hover{
  border: 0.1rem solid;
  border-image-slice: 1;
  border-width: 0.5rem;
  border-image-source: linear-gradient(to left, #090213, #078a85);
  }
`;
