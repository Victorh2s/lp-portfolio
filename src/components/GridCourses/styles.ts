import { Wrapper as TextGrid } from '../TextGrid/styles';
import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';
import { Certification } from '../Certification/styles';

export const Grid = styled.main`
  ${({ theme }) => css`

  @media ${theme.media.lteMedium}{
    ${Heading}{
      font-size: ${theme.font.sizes.medium};
    }

}
  `}
`;

export const Container = styled.section`
  ${({ theme }) => css`
  border-bottom: 0.1rem solid ${theme.colors.green};

  ${TextGrid}{
    margin-top: -0.1rem;
  }

  @media ${theme.media.lteMedium}{
    ${TextGrid}{
    font-size: 1.4rem;
  }

}

  `}
`;

export const Section = styled.div`
  ${({ theme }) => css`
  display: flex;


  @media ${theme.media.lteMedium}{
      display: block;
      align-items:center;
      ${Certification}{
        margin-top: -0.5rem;
        margin-bottom: 1rem;
        padding: 0;
      }

  }

  `}
`;

export const Title = styled.div`
  ${({ theme }) => css`

  display: flex;
  align-items: center;

  >svg {
    width: 5rem;
    height: 5rem;
    color: ${theme.colors.white};
  }

  @media ${theme.media.lteMedium}{
    ${Heading}{
      font-size: 2rem;
    }

}


  `}
`;
