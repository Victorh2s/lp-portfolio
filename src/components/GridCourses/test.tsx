import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { GridCourses } from '.';

describe('<GridCourses />', () => {
  it('should render the components', () => {
    renderTheme(<GridCourses />);

    expect(screen.getAllByLabelText('udemy')).toHaveLength(3);
    expect(
      screen.getByRole('heading', { name: 'Licen√ßas e certificados' }),
    ).toBeInTheDocument();

    expect(
      screen.getByText(/Curso de JavaScript e TypeScript/i),
    ).toBeInTheDocument();
  });
});
