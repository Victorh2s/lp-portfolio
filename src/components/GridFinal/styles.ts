import styled, { css } from 'styled-components';
import { Wrapper as TextGrid } from '../TextGrid/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`

  display: flex;
  padding: ${theme.spacings.xlarge};
  align-items: center;
  justify-content: center;
  border: 0.1rem solid;
  border-image-slice: 1;
  border-width: 0.1rem;
  border-image-source: linear-gradient(to top, #078a85, #090213);


  @media ${theme.media.lteMedium}{
      display: block;
      align-items: center;

      ${Image}{
        width: 7rem;
        height: 7rem;


      }
  }
  `}
`;

export const container = styled.div`
  ${({ theme }) => css`
  padding: ${theme.spacings.xlarge};
  text-align: center;
  justify-content: center;

  ${TextGrid}{
  margin-bottom: -0.1rem;
  border: 0.1rem solid;
  border-image-slice: 1;
  border-width: 0.1rem;
  border-image-source: linear-gradient(to bottom, #090213, #078a85);
}

>svg{
  color:${theme.colors.white};
  width:10rem;
  height:10rem;
  border: 0.1rem solid;
  border-image-slice: 1;
  border-width: 0.1rem;
  border-image-source: linear-gradient(to bottom, #090213, #078a85);
  transition: all 200ms ease-in-out;
  padding: ${theme.spacings.small};
  cursor: pointer;

  &:hover{
  border: 0.1rem solid;
  border-image-slice: 1;
  border-width: 0.4rem;
  border-image-source: linear-gradient(to top, #090213, #078a85);
  }
}

@media ${theme.media.lteSmall}{
    ${TextGrid}{
    border: 0.1rem solid;
  border-image-slice: 1;
  border-width: 0.1rem;
  border-image-source: linear-gradient(to left, #090213, #078a85);
}
  }


  `}
`;

export const Image = styled.img`
  width:10rem;
  height:10rem;
  border: 0.1rem solid;
  border-image-slice: 1;
  border-width: 0.2rem;
  border-image-source: linear-gradient(to bottom, #090213, #078a85);
  transition: all 200ms ease-in-out;
  cursor: pointer;

  &:hover{
  border: 0.1rem solid;
  border-image-slice: 1;
  border-width: 0.5rem;
  border-image-source: linear-gradient(to top, #090213, #078a85);
  }
`;
