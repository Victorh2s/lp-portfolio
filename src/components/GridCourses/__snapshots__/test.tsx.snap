// Vitest Snapshot v1

exports[`<GridCourses /> > should match to snapshot 1`] = `
.c12 {
  color: #FFFFFF;
  font-size: 1.6rem;
  font-family: 'Work Sans',sans-serif;
  padding: 0;
}

.c2 {
  color: #FFFFFF;
  font-size: 3.2rem;
  text-transform: none;
}

.c7 {
  color: #FFFFFF;
  font-size: 2.4rem;
  text-transform: none;
}

.c9 {
  padding: 3.2rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: 'Work Sans',sans-serif;
  margin-top: 0.6rem;
}

.c10 >svg {
  width: 2rem;
  height: 2rem;
  color: #078a85;
}

.c3 {
  border-bottom: 0.1rem solid #078a85;
}

.c3 .c11 {
  margin-top: -0.1rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 >svg {
  width: 5rem;
  height: 5rem;
  color: #FFFFFF;
}

.c6 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

@media (max-width:768px) {
  .c9 > a {
    font-size: 1.4rem;
  }
}

@media (max-width:768px) {
  .c0 .c1 {
    font-size: 2.4rem;
  }
}

@media (max-width:768px) {
  .c3 .c11 {
    font-size: 1.4rem;
  }
}

@media (max-width:768px) {
  .c4 {
    display: block;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }

  .c4 .c8 {
    margin-top: -0.5rem;
    margin-bottom: 1rem;
    padding: 0;
  }
}

@media (max-width:768px) {
  .c5 .c1 {
    font-size: 2rem;
  }
}

<div>
  <main
    class="c0"
  >
    <h4
      class="c1 c2"
    >
      Licen√ßas e certificados
    </h4>
    <section
      class="c3"
    >
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <svg
            aria-hidden="true"
            aria-label="udemy"
            class="c6"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 0 5.81 3.573v3.574l6.189-3.574 6.191 3.574V3.573zM5.81 10.148v8.144c0 1.85.589 3.243 1.741 4.234S10.177 24 11.973 24s3.269-.482 4.448-1.474c1.179-.991 1.768-2.439 1.768-4.314v-8.064h-3.242v7.85c0 2.036-1.509 3.055-2.948 3.055-1.428 0-2.947-.991-2.947-3.027v-7.878z"
            />
          </svg>
          <h5
            class="c1 c7"
          >
            Udemy
          </h5>
        </div>
        <div
          class="c8 c9"
        >
          <a
            class="c10"
            href="https://www.udemy.com/certificate/UC-fcc76692-e6f8-4c7a-a71b-7e36dce27413/"
            target="_blank"
          >
            Exibir Credencial 
            <svg
              aria-hidden="true"
              aria-label="icon"
              class="c6"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m13 3 3.293 3.293-7 7 1.414 1.414 7-7L21 11V3z"
              />
              <path
                d="M19 19H5V5h7l-2-2H5c-1.103 0-2 .897-2 2v14c0 1.103.897 2 2 2h14c1.103 0 2-.897 2-2v-5l-2-2v7z"
              />
            </svg>
          </a>
        </div>
      </div>
      <p
        class="c11 c12"
      >
        Curso de JavaScript e TypeScript.
      </p>
    </section>
    <section
      class="c3"
    >
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <svg
            aria-hidden="true"
            aria-label="udemy"
            class="c6"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 0 5.81 3.573v3.574l6.189-3.574 6.191 3.574V3.573zM5.81 10.148v8.144c0 1.85.589 3.243 1.741 4.234S10.177 24 11.973 24s3.269-.482 4.448-1.474c1.179-.991 1.768-2.439 1.768-4.314v-8.064h-3.242v7.85c0 2.036-1.509 3.055-2.948 3.055-1.428 0-2.947-.991-2.947-3.027v-7.878z"
            />
          </svg>
          <h5
            class="c1 c7"
          >
            Udemy
          </h5>
        </div>
        <div
          class="c8 c9"
        >
          <a
            class="c10"
            href="https://www.udemy.com/certificate/UC-fcc76692-e6f8-4c7a-a71b-7e36dce27413/"
            target="_blank"
          >
            Exibir Credencial 
            <svg
              aria-hidden="true"
              aria-label="icon"
              class="c6"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m13 3 3.293 3.293-7 7 1.414 1.414 7-7L21 11V3z"
              />
              <path
                d="M19 19H5V5h7l-2-2H5c-1.103 0-2 .897-2 2v14c0 1.103.897 2 2 2h14c1.103 0 2-.897 2-2v-5l-2-2v7z"
              />
            </svg>
          </a>
        </div>
      </div>
      <p
        class="c11 c12"
      >
        Testando aqui
      </p>
    </section>
    <section
      class="c3"
    >
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <svg
            aria-hidden="true"
            aria-label="udemy"
            class="c6"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 0 5.81 3.573v3.574l6.189-3.574 6.191 3.574V3.573zM5.81 10.148v8.144c0 1.85.589 3.243 1.741 4.234S10.177 24 11.973 24s3.269-.482 4.448-1.474c1.179-.991 1.768-2.439 1.768-4.314v-8.064h-3.242v7.85c0 2.036-1.509 3.055-2.948 3.055-1.428 0-2.947-.991-2.947-3.027v-7.878z"
            />
          </svg>
          <h5
            class="c1 c7"
          >
            Udemy
          </h5>
        </div>
        <div
          class="c8 c9"
        >
          <a
            class="c10"
            href="https://www.udemy.com/certificate/UC-fcc76692-e6f8-4c7a-a71b-7e36dce27413/"
            target="_blank"
          >
            Exibir Credencial 
            <svg
              aria-hidden="true"
              aria-label="icon"
              class="c6"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m13 3 3.293 3.293-7 7 1.414 1.414 7-7L21 11V3z"
              />
              <path
                d="M19 19H5V5h7l-2-2H5c-1.103 0-2 .897-2 2v14c0 1.103.897 2 2 2h14c1.103 0 2-.897 2-2v-5l-2-2v7z"
              />
            </svg>
          </a>
        </div>
      </div>
      <p
        class="c11 c12"
      >
        Testando aqui
      </p>
    </section>
  </main>
</div>
`;
